File: leedcode/3_longestSubstringLength.cpp
#include <vector>
#include <string>
int lengthOfLongestSubstring(string s)
{
    vector<int> dic(256,-1);
    int start = maxLen = 0;
    for(int i = 0; i < strlen(s); ++i)
    {
        if(dic[s[i]] >= start)
            start = dic[s[i]] + 1;
        maxLen = max(maxLen,i - start + 1);
        dic[s[i]] = i;
    }
    return maxLen;
}

File: leedcode/3_longestSubstringLength.py
##最长不重复连续子串的大小
def lengthOfLongestString(s):
	start = maxlen = 0
	dic = {}
	for i in range(len(s)):
		if s[i] in dic and start <= dic[s[i]]:
			start = dic[s[i]] + 1
	    maxlen = max(maxlen,i - start + 1)
	    dic[s[i]] = i
	return maxlen
